kernel/default.nix

```diff
@@ -0,0 +1,105 @@
+{
+  pkgs ? import (fetchTarball {
+    url = "https://github.com/NixOS/nixpkgs/archive/refs/tags/24.05.tar.gz";
+    sha256 = "sha256:1lr1h35prqkd1mkmzriwlpvxcb34kmhc9dnr48gkm8hh089hifmx";
+  }) {}
+}:
+let
+  arch = pkgs.stdenv.hostPlatform.uname.processor;
```

Comment:
may be it's a not top priority right now, but I would also go for cross-compilation later

Reply:
I was thinking about cross compilation as well and that would be ideal. One big questionmark I see there is that the binaries (one cross-compiled, one natively built) will likely not be identical as the toolchain for building is not the exact same. If that can be solved I am all for it.